src/reachy2_sdk/parts/arm.py

```diff
@@ -394,11 +394,28 @@ def _goto_cartesian_interpolation(
             request = ArmCartesianGoal(
                 id=self._part_id,
                 goal_pose=Matrix4x4(data=interpolated_matrix.flatten().tolist()),
-                duration=FloatValue(value=time_step),
             )
             self._arm_stub.SendArmCartesianGoal(request)
             time.sleep(time_step)
 
+        current_pose = self.forward_kinematics()
+        precision_distance_xyz = np.linalg.norm(current_pose[:3, 3] - target[:3, 3])
+        if precision_distance_xyz > 0.003:
+            print("Precision is not good enough, spamming the goal position!")
+            for t in np.linspace(0, 1, nb_steps):
+                # Spamming the goal position to make sure its reached
+                request = ArmCartesianGoal(
+                    id=self._part_id,
+                    goal_pose=Matrix4x4(data=target.flatten().tolist()),
+                )
+                self._arm_stub.SendArmCartesianGoal(request)
+                time.sleep(time_step)
+
+            time.sleep(0.1)
```

Comment:
This time.sleep is weird (I know, I'm the one who added it).
Maybe we should add a comment? Such as:
Small delay to make sure the present_position is correctly read